src/eos-updater-apply.c

```diff
@@ -120,42 +161,74 @@ apply (GTask *task,
     {
       g_autofree gchar *rev = NULL;
 
-      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, &error))
-        goto error;
+      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, error))
+        return FALSE;
 
       if (rev)
         {
-          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, error))
+            return FALSE;
 
           ostree_repo_transaction_set_refspec (repo, orig_refspec, NULL);
 
-          if (!ostree_repo_commit_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_commit_transaction (repo, NULL, cancel, error))
+            return FALSE;
         }
     }
 
   if (!ostree_sysroot_simple_write_deployment (sysroot,
-                                               NULL,
+                                               osname,
                                                new_deployment,
-                                               merge_deployment,
+                                               booted_deployment,
                                                0,
                                                cancel,
-                                               &error))
-    goto error;
+                                               error))
+    return FALSE;
 
   newbootver = ostree_deployment_get_deployserial (new_deployment);
 
-  g_task_return_boolean (task, bootversion != newbootver);
-  goto cleanup;
+  if (!eos_extensions_save (data->extensions,
+                            repo,
+                            cancel,
+                            error))
+    return FALSE;
+  g_clear_object (&data->extensions);
+
+  head_commit_timestamp = g_date_time_new_from_unix_utc (ostree_commit_get_timestamp (update_commit));
+
+  if (!eos_avahi_generate_service_file (repo,
+                                        head_commit_timestamp,
+                                        error))
+    return FALSE;
```

Comment:
I don't think that failures in `eos_extensions_save` or `eos_avahi_generate_service_file` should propagate here. This code is about whether the update was applied, not whether it could be served to someone else. A failure from this function returns to the user saying that their system could not be upgraded, which is not the case here. I'd turn these into warnings or find some other way to hook into the upgrade path.